app-misc/lcd4linux/lcd4linux-0.11.0_pre20170527.ebuild

```diff
@@ -0,0 +1,165 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 )
+
+inherit autotools flag-o-matic python-single-r1
+
+DESCRIPTION="A small program that grabs information and displays it on an external LCD"
+HOMEPAGE="https://lcd4linux.bulix.org/"
+SRC_URI="https://www.bl4ckb0x.de/files/${P}.tar.bz2"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dmalloc outb"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )
+	lcd_devices_hd44780? ( !lcd_devices_hd44780-i2c )
+	lcd_devices_hd44780-i2c? ( !lcd_devices_hd44780 )"
+
+# Define the list of valid lcd devices.
+# Some drivers were removed from this list:
+# - lcdlinux: It's an ancient driver, which needs app-misc/lcd-linux, that made it never to the portage tree.
+# - lcdlinux: Besides, app-misc/lcd-linux won't compile on a modern linux kernel.
+# - st2205: It's needs dev-libs/libst2205,  which made it never to the portage tree and is quite outdated.
+IUSE_LCD_DEVICES=( astusb:ASTUSB beckmannegle:BeckmannEgle bwct:BWCT crystalfontz:CrystalFontz curses:Curses
+	cwlinux:Cwlinux d4d:D4D dpf:DPF ea232graphic:EA232graphic efn:EFN futabavfd:FutabaVFD fw8888:FW8888
+	g15:G15 glcd2usb:GLCD2USB hd44780:HD44780 hd44780-i2c:HD44780-I2C irlcd:IRLCD lcd2usb:LCD2USB
+	lcdterm:LCDTerm ledmatrix:LEDMatrix lph7508:LPH7508 luise:LUIse	lwabp:LW_ABP m50530:M50530
+	matrixorbital:MatrixOrbital matrixorbitalgx:MatrixOrbitalGX mdm166a:MilfordInstruments
+	milfordinstruments:MDM166A newhaven:Newhaven noritake:Noritake null:NULL pertelian:Pertelian
+	phanderson:PHAnderson picgraphic:PICGraphic picolcd:picoLCD picolcdgraphic:picoLCDGraphic png:PNG
+	ppm:PPM routerboard:RouterBoard sample:Sample samsungspf:SamsungSPF serdisplib:serdisplib
+	shuttlevfd:ShuttleVFD simplelcd:SimpleLCD t6963:T6963 teaklcm:TeakLCM trefon:Trefon ula200:ULA200
+	usbhub:USBHUB usblcd:USBLCD vnc:VNC wincornixdorf:WincorNixdorf x11:X11 )
+
+# Define the list of valid lcd4linux plugins.
+# Some plugins were removed from this list:
+# - imon: Uses telmond, which is only available on a fli4l router or an eisfair server.
+# - ppp: It has been replaced by the netdev plugin.
+# - seti: SETI@home software was replaced by sci-misc/boinc, which is not compatible.
+# - xmms: XMMS software was replaced by media-sound/xmms2, which is not compatible.
+IUSE_LCD4LINUX_PLUGINS=( apm asterisk button_exec cpuinfo dbus diskstats dvb exec event
+	fifo file gps hddtemp huawei i2c_sensors iconv isdn kvv
+	loadavg meminfo mpd mpris_dbus mysql netdev netinfo pop3
+	proc_stat python qnaplog raspi sample statfs uname uptime
+	w1retap wireless )
+
+# Add supported drivers from 'IUSE_LCD_DEVICES' to 'IUSE' and 'REQUIRED_USE'
+REQUIRED_USE+=" || ( "
+for lcd_device in ${IUSE_LCD_DEVICES[@]}; do
+	IUSE+=" lcd_devices_${lcd_device%%:*} "
```

Comment:
So if you want to play with it further, you can do:

    lcd_device=${lcd_device,,}

to make it lowercase. In that case, you wouldn't have to repeat the names if flag == lowercase(upstream-name), i.e. you'd have:

    IUSE_LCD_DEVICES=( ASTUSB BeckmannEgle BWCT ... )

Reply:
And if you don't want to do that, you can avoid the loop by inverting the order, and doing:

    IUSE+=" ${IUSE_LCD_DEVICES[@]/*:/lcd_devices_}"

Reply:
Thanks, added the first variant. But the second one `IUSE+=" ${IUSE_LCD_DEVICES[@]/*:/lcd_devices_}"`seems not to work for me. But I don't even understand it fully.